main/main.cpp

```diff
@@ -654,6 +655,7 @@ void Main::print_help(const char *p_binary) {
 	print_help_option("-s, --script <script>", "Run a script.\n");
 	print_help_option("--main-loop <main_loop_name>", "Run a MainLoop specified by its global class name.\n");
 	print_help_option("--check-only", "Only parse for errors and quit (use with --script).\n");
+	print_help_option("--format", "Format the file in-place or (if used with --check-only) check if file is formatted correctly. This option must be used with --script.\n");
```

Comment:
The --script option is script language agnostic. It either needs to be documented that `--format` is gdscript specific (in this case a check whether the script actually is GDScript would be nice). Or this should be implemented using the ScriptLanguage interface to keep it script language agnostic as well.

(Given the current feature set of the CLI argument, we could also remove it for now and add it in a followup)

Reply:
Good point - I've documented that, left a TODO, and added a check for now as I'd like to add CLI interface within this PR. The reason is that CLI is the fastest way to manually test formatter - it doesn't require linking tests and can be run headless.